classes/Category.php

```diff
@@ -356,7 +356,14 @@ public function delete()
             Category::regenerateEntireNtree();
         }
 
-        Hook::exec('actionCategoryDelete', array('category' => $this));
+        $deleted_children = array();
+        foreach ($all_cat as $cat) {
+            if ($cat != $this) {
```

Comment:
You want to remove $this in the array. Just call $deleted_children; with my proposal just above instead of parsing the full array.